### PR TITLE
Fix string index out of bounds crash.

### PR DESCRIPTION
**Pull request by RoyalPineapple**
Well, this is always embarrassing.

`string.startIndex...string.endIndex` is out of bounds, which seems counter to my understanding of either ranges or string indices but I'm not sure which . 

Regardless, we need a half open range here.

### CODE REVIEW

File: CHANGELOG.md
Comment (kyleve) [inline CHANGELOG.md:1]: Tests?

Comment (RoyalPineapple) [reply CHANGELOG.md:1]: yep

File: BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift [hunk 1]
```diff
@@ -357,6 +357,70 @@ class AttributedLabelTests: XCTestCase {
         compareSnapshot(of: element)
     }
 
+    func test_linkAccessibility() {
+        let labelview = AttributedLabel.LabelView()
+
+        do {
+            // Test that link insertion happy path works
```
Comment (kyleve) [inline BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift:364]: Can we add a test for a link at the very end of the string too?

**Review by kyleve — APPROVED** [review]

File: BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift (hunk 1 above)
Comment (RoyalPineapple) [reply BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift:364]: done

File: BlueprintUICommonControls/Sources/AttributedLabel.swift [hunk 2]
```diff
@@ -565,17 +565,32 @@ extension AttributedLabel {
             // • Square brackets aren't often used in prose, unlike parenthesis. They're unlikely to be confused with the actual content.
             // • They look like markdown.
 
-            let insertionString = "[\(localizedLinkString)] "
+            let insertionString = "[\(localizedLinkString)]"
             // Insert from the end of the string to keep indices stable.
             let reversed = links.sorted { $0.range.location > $1.range.location }
             for link in reversed {
-                let insertionPoint = label.index(label.startIndex, offsetBy: link.range.location + link.range.length)
+                // Extract substring from NSString to allign with NSRange provided by the link.
```
Comment (kyleve) [inline BlueprintUICommonControls/Sources/AttributedLabel.swift:572]: `allign` -> `align`

File: BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift [hunk 3]
```diff
@@ -357,6 +357,109 @@ class AttributedLabelTests: XCTestCase {
...
+                    in: string as String,
+                    linkAccessibilityLabel: "."
+                )
+                XCTAssertEqual(accessibilityLabel, string)
+            }
+        }
+
+        do {
+            // Test with emoji
+            let string = "🇺🇸🇨🇦🇯🇵🇫🇷"
```
Comment (kyleve) [inline BlueprintUICommonControls/Tests/Sources/AttributedLabelTests.swift:425]: These are great, thanks for adding them!

File: BlueprintUICommonControls/Sources/AttributedLabel.swift (hunk 2 above)
Comment (RoyalPineapple) [reply BlueprintUICommonControls/Sources/AttributedLabel.swift:572]: done

**Comment by robmaceachern** [comment]
> `string.startIndex...string.endIndex` is out of bounds, which seems counter to my understanding of either ranges or string indices but I'm not sure which

Ah yeah it's the `endIndex` bit.

> [endIndex](https://developer.apple.com/documentation/swift/string/endindex)
> A string’s “past the end” position—that is, the position one greater than the last valid subscript argument.

Guess they should have called it `pastTheEndIndex` 😛

**Pull request merged**